pkg/config/json_test.go

```diff
@@ -330,6 +330,44 @@ type jsonStabilityTest struct {
 	expectedJSON  string
 }
 
+func testVirtioNetVfkitMagicJson(t *testing.T, vfkitMagic bool) {
+	jsonTemplate := `{"vcpus":3,"memoryBytes":4194304000,"bootloader":{"kind":"linuxBootloader","vmlinuzPath":"/vmlinuz","initrdPath":"/initrd","kernelCmdLine":"console=hvc0"},"devices":[{"kind":"virtionet","nat":false,"unixSocketPath":"/some/path/to/socket","macAddress":"00:11:22:33:44:55"%s}]}`
+	var jsonStr string
+	var unmarshalledVM VirtualMachine
+
+	if vfkitMagic {
+		jsonStr = fmt.Sprintf(jsonTemplate, ``)
+	} else {
+		jsonStr = fmt.Sprintf(jsonTemplate, `,"vfkitMagic":false`)
+	}
+	err := json.Unmarshal([]byte(jsonStr), &unmarshalledVM)
+	require.NoError(t, err)
+	netDevs := unmarshalledVM.VirtioNetDevices()
+	require.Len(t, netDevs, 1)
+	require.Equal(t, netDevs[0].VfkitMagic, vfkitMagic)
+
+	vm := newLinuxVM(t)
+	dev, err := VirtioNetNew("00:11:22:33:44:55")
+	require.NoError(t, err)
+	dev.SetUnixSocketPath("/some/path/to/socket")
+	if !vfkitMagic {
+		dev.VfkitMagic = false
+	}
```

Comment:
Adding if blocks in a test like this make the tests harder to follow and easier to get wrong.

Reply:
I tried to do this differently, let me know if you feel it’s better.